packages/example/pages/index.tsx

```diff
@@ -159,95 +160,132 @@ const Example = ({ authEnabled }: AppContextProps) => {
+                                height: 12,
+                                marginRight: 4,
+                                overflow: 'hidden',
+                                width: 12,
+                              }}
+                            >
+                              {chain.iconUrl && (
+                                <img
+                                  alt={chain.name ?? 'Chain icon'}
+                                  src={chain.iconUrl}
+                                  style={{ height: 12, width: 12 }}
+                                />
+                              )}
+                            </div>
+                          )}
+                          {chain.name ?? chain.id}
+                        </button>
+
+                        <button onClick={openAccountModal} type="button">
+                          {account.displayName}
+                          {account.displayBalance
+                            ? ` (${account.displayBalance})`
+                            : ''}
+                        </button>
+                      </div>
                     );
-                  }
+                  })()}
+                </div>
+              );
+            }}
+          </ConnectButton.Custom>
 
-                  return (
-                    <div style={{ display: 'flex', gap: 12 }}>
-                      <button
-                        onClick={openChainModal}
-                        style={{ alignItems: 'center', display: 'flex' }}
-                        type="button"
-                      >
-                        {chain.hasIcon && (
-                          <div
-                            style={{
-                              background: chain.iconBackground,
-                              borderRadius: 999,
-                              height: 12,
-                              marginRight: 4,
-                              overflow: 'hidden',
-                              width: 12,
-                            }}
-                          >
-                            {chain.iconUrl && (
-                              <img
-                                alt={chain.name ?? 'Chain icon'}
-                                src={chain.iconUrl}
-                                style={{ height: 12, width: 12 }}
-                              />
-                            )}
-                          </div>
-                        )}
-                        {chain.name ?? chain.id}
-                      </button>
+          <RainbowButton.Custom>
+            {({ ready, connect }) => {
+              return (
+                <button
+                  type="button"
+                  disabled={!ready}
+                  onClick={connect}
+                  style={{ marginLeft: '18px' }}
+                >
+                  Connect Rainbow
+                </button>
+              );
+            }}
+          </RainbowButton.Custom>
+        </div>
```

Comment:
We should add a custom button for the metamask wallet button here too

Reply:
Great idea. Yeah that would be awesome to include custom connectors 🎶 🙌